pkg/test/nstemplateset/nstemplateset_assertion_test.go

```diff
@@ -0,0 +1,115 @@
+package nstemplateset_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test/nstemplateset"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stretchr/testify/assert"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestNSTemplateSetAssertion(t *testing.T) {
+
+	t.Run("check space roles", func(t *testing.T) {
+		// given
+		nsTmplSet := nstemplateset.NewNSTemplateSet("foo")
+		nsTmplSet.Spec.SpaceRoles = []toolchainv1alpha1.NSTemplateSetSpaceRole{
+			{
+				TemplateRef: "base-admin-123456b",
+				Usernames:   []string{"admin1", "admin2"},
+			},
+			{
+				TemplateRef: "base-viewer-123456b",
+				Usernames:   []string{"viewer1"},
+			},
+		}
+		cl := test.NewFakeClient(t, nsTmplSet)
+		cl.MockGet = func(ctx context.Context, key types.NamespacedName, obj runtimeclient.Object) error {
+			if key.Namespace == test.MemberOperatorNs && key.Name == "foo" {
+				if obj, ok := obj.(*toolchainv1alpha1.NSTemplateSet); ok {
+					*obj = *nsTmplSet
+					return nil
+				}
+			}
+			return fmt.Errorf("unexpected object key: %v", key)
+		}
+
+		t.Run("success", func(t *testing.T) {
+			// given
+			mockT := test.NewMockT()
+			// when
+			nstemplateset.AssertThatNSTemplateSet(mockT, nsTmplSet.Namespace, nsTmplSet.Name, cl).HasSpaceRoles(
+				nstemplateset.SpaceRole("base-admin-123456b", "admin1", "admin2"),
+				nstemplateset.SpaceRole("base-viewer-123456b", "viewer1"),
+			)
+			// then: all good
+			assert.False(t, mockT.CalledFailNow())
+			assert.False(t, mockT.CalledFatalf())
+			assert.False(t, mockT.CalledErrorf()) // called
+		})
+
+		t.Run("fails when actual user is not expected", func(t *testing.T) {
+			// given
+			mockT := test.NewMockT()
+			// when
+			nstemplateset.AssertThatNSTemplateSet(mockT, test.MemberOperatorNs, "foo", cl).HasSpaceRoles(
+				nstemplateset.SpaceRole("base-admin-123456b", "admin1"), // admin2 is not expected
+				nstemplateset.SpaceRole("base-viewer-123456b", "viewer1"),
+			)
+			// then: all good
```

Comment:
I guess this comment can be removed for all the error scenarios?

Reply:
may I ask why? the "all good" comment is to explain that no error or failure should have been reported by the assertion

Reply:
ahh, I see. I did not realize you were pointing to a case where an error is expected! 🤦‍♂️ 
let me fix that in a subsequent PR :)

Reply:
see https://github.com/codeready-toolchain/toolchain-common/pull/259